pageserver/src/http/routes.rs

```diff
@@ -1561,7 +1501,20 @@ async fn timeline_collect_keyspace(
             .await
             .map_err(|e| ApiError::InternalServerError(e.into()))?;
 
-        json_response(StatusCode::OK, Partitioning { keys, at_lsn })
+        let res = crate::http::models::partitioning::Partitioning { keys, at_lsn };
+        if check_serialization_roundtrip {
+            (|| {
+                let ser = serde_json::ser::to_vec(&res).context("serialize")?;
+                let de: crate::http::models::partitioning::Partitioning =
+                    serde_json::from_slice(&ser).context("deserialize")?;
+                anyhow::ensure!(de == res, "not equal");
+                info!("passed serialization rountrip check");
+                Ok(())
+            })()
+            .context("serialization rountrip")
+            .map_err(ApiError::InternalServerError)?;
+        }
+        json_response(StatusCode::OK, res)
```

Comment:
Wondering whether it's easier to write a Rust unit test that uses TenantHarness, vs add a python test that loads up a tenant and then calls this API. Thoughts?

Reply:
The serialization roundtrip part definitely belongs in a rust unit test.

For checking the actual HTTP endpoint, I would lean toward just writing a python test to avoid having to write the harness for instantiating the HTTP API stuff in a rust unit test.  But if that harness already existed I would be fine with the rust unit test, for simple things where we're just checking a GET works.

Reply:
moved to https://github.com/neondatabase/neon/pull/6145/files#r1427988898